cmd/loadtest/server.go

```diff
@@ -0,0 +1,162 @@
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	newSimpleController := func(id int, status chan<- control.UserStatus) control.UserController {
+		ueConfig := userentity.Config{
+			ServerURL:    config.ConnectionConfiguration.ServerURL,
+			WebSocketURL: config.ConnectionConfiguration.WebSocketURL,
+		}
+		ue := userentity.New(memstore.New(), ueConfig)
+		return simplecontroller.New(id, ue, status)
+	}
+
+	lt := loadtest.New(&config, newSimpleController)
+	a.loadTests[lt.Id] = lt
+	// start := time.Now()
+	// err = lt.Run()
+	// if err != nil {
+	// 	return err
+	// }
+
+	// mlog.Info("loadtest started")
+	// time.Sleep(60 * time.Second)
+
+	// err = lt.Stop()
+	// mlog.Info("loadtest done", mlog.String("elapsed", time.Since(start).String()))
+
+	// return err
+
+	w.Header().Set("Content-Type", "application/json")
+	_ = json.NewEncoder(w).Encode(map[string]string{"loadTestId": lt.Id})
+
+}
+
+func (a *Agent) getLoadTestById(w http.ResponseWriter, r *http.Request) (*loadtest.LoadTester, error) {
+	vars := mux.Vars(r)
+	id := vars["id"]
+	lt, ok := a.loadTests[id]
+	if !ok {
+		err := fmt.Errorf("Load test with id %s not found", id)
+		http.Error(w, err.Error(), http.StatusNotFound)
+		return nil, err
+	}
+	return lt, nil
+}
+func (a *Agent) runLoadTestHandler(w http.ResponseWriter, r *http.Request) {
+	lt, err := a.getLoadTestById(w, r)
+	if err != nil {
+		return
+	}
+	go lt.Run()
+	w.Header().Set("Content-Type", "application/json")
+	_ = json.NewEncoder(w).Encode(map[string]string{"message": "load test started"})
+}
+
+func (a *Agent) stopLoadTestHandler(w http.ResponseWriter, r *http.Request) {
+	lt, err := a.getLoadTestById(w, r)
+	if err != nil {
+		return
+	}
+	_ = lt.Stop()
+	w.Header().Set("Content-Type", "application/json")
+	_ = json.NewEncoder(w).Encode(map[string]string{"message": "load test stopped"})
+}
+
+func (a *Agent) destroyLoadTestHandler(w http.ResponseWriter, r *http.Request) {
+	lt, err := a.getLoadTestById(w, r)
+	if err != nil {
+		return
+	}
+	_ = lt.Stop()
+	w.Header().Set("Content-Type", "application/json")
+	_ = json.NewEncoder(w).Encode(map[string]string{"message": "load test destroyed"})
+	delete(a.loadTests, lt.Id)
+}
+
+func (a *Agent) getLoadTestStatusHandler(w http.ResponseWriter, r *http.Request) {
+	lt, err := a.getLoadTestById(w, r)
+	if err != nil {
+		return
+	}
+	w.Header().Set("Content-Type", "application/json")
+	_ = json.NewEncoder(w).Encode(map[string]loadtest.State{"status": lt.GetState()})
+}
+
+func (a *Agent) addUserHandler(w http.ResponseWriter, r *http.Request) {
+	lt, err := a.getLoadTestById(w, r)
+	if err != nil {
+		return
+	}
+	lt.AddUser()
+	w.Header().Set("Content-Type", "application/json")
+	_ = json.NewEncoder(w).Encode(map[string]string{"message": "user added"})
+
+}
+
+func (a *Agent) removeUserHandler(w http.ResponseWriter, r *http.Request) {
+	lt, err := a.getLoadTestById(w, r)
+	if err != nil {
+		return
+	}
+	lt.RemoveUser()
```

Comment:
Same as above. Need to support removing `n` users where `n` is a value passed in the request.